app/code/community/SomethingDigital/SearchPerf/Trait/ResetsSearchResults.php

```diff
@@ -14,14 +14,42 @@
      */
     protected function _resetSearchResults()
     {
-        $adapter = $this->_getWriteAdapter();
-        $adapter->update($this->_getTable('catalogsearch/search_query'), array('is_processed' => 0), array('is_processed != 0'));
-        $adapter->delete($this->_getTable('catalogsearch/result'));
-
+        $this->_doResetQueries();
         $this->_app->dispatchEvent('enterprise_catalogsearch_reset_search_result', array());
     }
 
     /**
+     * Perform actual reset queries.
+     *
+     * @return void
+     */
+    protected function _doResetQueries()
+    {
+        if (!$this->shouldSkipResetQueries()) {
```

Comment:
Interesting. Didn't know the changed the internals like [this](https://github.com/OpenMage/magento-mirror/blob/magento-1.9/app/code/core/Mage/CatalogSearch/Model/Resource/Fulltext.php#L300-L304)